src/Horarium.InMemory/InMemoryRepository.cs

```diff
@@ -0,0 +1,156 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Horarium.Repository;
+
+namespace Horarium.InMemory
+{
+    public class InMemoryRepository : IJobRepository
+    {   
+        private readonly ConcurrentDictionary<string, JobInMemoryWrapper> _jobsStorage = 
+            new ConcurrentDictionary<string, JobInMemoryWrapper>();
+        
+        private readonly ConcurrentDictionary<string, RecurrentJobSettings> _recurrentJobSettingsStorage = 
+            new ConcurrentDictionary<string, RecurrentJobSettings>();
+        
+        public Task<JobDb> GetReadyJob(string machineName, TimeSpan obsoleteTime)
+        {
+            bool Filter(JobDb x) =>
+                (x.Status == JobStatus.Ready || x.Status == JobStatus.RepeatJob) &&
+                x.StartAt < DateTime.UtcNow ||
+                x.Status == JobStatus.Executing && x.StartedExecuting < DateTime.UtcNow - obsoleteTime;
+
+            void Update(JobDb x)
+            {
+                x.Status = JobStatus.Executing;
+                x.ExecutedMachine = machineName;
+                x.StartedExecuting = DateTime.UtcNow;
+                x.CountStarted++;
+            }
+
+            return Task.FromResult(UpdateOneAtomic(Filter, Update));
+        }
+
+        public Task AddJob(JobDb job)
+        {
+            var wrappedJob = JobInMemoryWrapper.CreateJobInMemoryWrapper(job);
+            _jobsStorage.AddOrUpdate(job.JobId, wrappedJob, (_, __) => wrappedJob);
```

Comment:
go style? Maybe use (x,y) better?

Reply:
I dont see much diference TBH, but if you insist - sure